jetty-util/src/main/java/org/eclipse/jetty/util/NanoTime.java

```diff
@@ -0,0 +1,180 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * <p>Utility class with methods that deal with {@link System#nanoTime()}.</p>
+ */
+public class NanoTime
+{
+    /**
+     * @return the current nanoTime via {@link System#nanoTime()}
+     */
+    public static long now()
+    {
+        return System.nanoTime();
+    }
+
+    /**
+     * <p>Calculates the nanoseconds elapsed between two nanoTimes.</p>
+     *
+     * @param beginNanoTime the begin nanoTime
+     * @param endNanoTime the end nanoTime
+     * @return the nanoseconds elapsed
+     */
+    public static long elapsed(long beginNanoTime, long endNanoTime)
+    {
+        return endNanoTime - beginNanoTime;
+    }
+
+    /**
+     * <p>Calculates the nanoseconds elapsed between a begin nanoTime and the current nanoTime.</p>
+     *
+     * @param beginNanoTime the begin nanoTime
+     * @return the nanoseconds elapsed between the given begin nanoTime and the current nanoTime
+     */
+    public static long elapsedFrom(long beginNanoTime)
+    {
+        return elapsed(beginNanoTime, now());
+    }
+
+    /**
+     * <p>Calculates the nanoseconds elapsed between the current nanoTime and an end nanoTime.</p>
+     *
+     * @param endNanoTime the end nanoTime
+     * @return the nanoseconds elapsed between the current nanoTime and the given end nanoTime
+     */
+    public static long elapsedTo(long endNanoTime)
```

Comment:
This name is confusing as `endNanoTime` is supposed to be in the future but the past tense in the method name suggests that it's in the past.
Something like `remaningTo` for instance sounds better to me.

Reply:
I like `remainingTo`.

Reply:
`remainingTo` is not good phrasing.  It should be `remainingUntil` or just `until`.   I think using `elapsed`, `since` and `until` is clear.